lib/timescaledb/connection_handling.rb

```diff
@@ -5,12 +5,25 @@ class ConnectionNotEstablishedError < StandardError; end
   
   # @param [String] config with the postgres connection string.
   def establish_connection(config)
+    # Establish connection for Timescaledb
     Connection.instance.config = config
+    
+    # Also establish connection for ActiveRecord if it's defined
+    if defined?(ActiveRecord::Base)
+      ActiveRecord::Base.establish_connection(config)
+    end
   end
 
   # @param [PG::Connection] to use it directly from a raw connection
-  def use_connection conn
-    Connection.instance.use_connection conn
+  def use_connection(conn)
+    Connection.instance.use_connection(conn)
+    
+    # Also set ActiveRecord connection if it's defined
+    if defined?(ActiveRecord::Base) && ActiveRecord::Base.connected?
+      # Get the existing connection handler and update it
+      ar_conn = ActiveRecord::Base.connection
```

Comment:
This [broken build](https://github.com/timescale/timescaledb-ruby/actions/runs/13379439798/job/37365267424)

Reply:
@jonatas Were you able to fix it?

Reply:
Yes, I don't understand why this was happening. This detection is also better if people can handle multiple libs interacting with the  DB. Do you see any problem with the implementation?